src/bootstrap/lib.rs

```diff
@@ -472,12 +472,22 @@ impl Build {
         slice::from_ref(&self.build.triple)
     }
 
+    /// If the LLVM submodule has been initialized already, sync it unconditionally. This avoids
+    /// contributors checking in a submodule change by accident.
+    pub fn maybe_update_llvm_submodule(&self) {
+        if self.config.src.join("src").join("llvm-project").join(".git").exists() {
+            native::update_llvm_submodule(&builder::Builder::new(self));
```

Comment:
nit: you don't actually need builder in this function, please just pass Build directly

Reply:
I do think I need the builder though? I use `builder.in_tree_llvm_info`, `builder.config`, `builder.run`.

Reply:
All of those are on Build. Builder is only necessary really if you're going to ensure(...) something, and we shouldn't be doing that here.

Reply:
Ahh got it, thank you!